File: Miscelaneous/produto.py
from dataclasses import dataclass


@dataclass
class Produto:
    nome: str
    preco: float
    quantidade_em_estoque: int

    def exibir_resumo(self):
        print(f"Produto: {self.nome} | R$ {self.preco:.2f} | estoque: {self.quantidade_em_estoque}")

File: Miscelaneous/Produto.java
public class Produto {
    private final String nome;
    private final double preco;
    private final int quantidadeEmEstoque;

    public Produto(String nome, double preco, int quantidadeEmEstoque) {
        this.nome = nome;
        this.preco = preco;
        this.quantidadeEmEstoque = quantidadeEmEstoque;
    }

    public void exibirResumo() {
        System.out.printf("Produto: %s | R$ %.2f | estoque: %d%n", nome, preco, quantidadeEmEstoque);
    }
}